msprime/ancestry.py

```diff
@@ -1698,7 +1705,38 @@ class DiracCoalescent(ParametricAncestryModel):
 @dataclasses.dataclass
 class SweepGenicSelection(ParametricAncestryModel):
     """
-    .. todo:: Document me
+    This simulates a coalescent history with a single selective
+    sweep that has occurred in the past. The model is one of a
+    a structured coalescent where selective backgrounds are defined as in
+    `Braverman et al. (1995) <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/>`_
+    The implementation details here follow closely to those in discoal,
+    `Kern and Schrider (2016)
+    <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5167068/>`_
+
+    .. warning::
+        As the strength of selection (:math:`2Ns`) increases the chance
+        of two coalescent events happening very very close together in
+        time and can lead to msprime errors. should you encounter this
```

Comment:
How about 

If the strength of selection  (:math:`2Ns`) is sufficiently large, the time difference between successive events can be smaller than the finite precision available, leading to zero length branches in the output trees. As this is not allowed by tskit, an error will be raised.

Reply:
is there a way to change this precision Jerome? I've been surprised that we get "time-travel" when nodes times should be ~1e-4 generations apart

Reply:
No, it's just plain ole fashioned IEEE 754 double precision like our Daddy's used. Sounds like a bug?